### PR TITLE
Update conventional station pressure (conv_ps) for testing and end-to-end

### PR DESCRIPTION
**Pull request by emilyhcliu**
This PR is to update the station pressure with the following modifications:

- Modify YAMLs under testing (UFO Evaluation)
   - Add error assignment by extracting errors from the GSI error table (needs an update from [UFO PR #3085)](https://github.com/JCSDA-internal/ufo/pull/3085)
   - Update options for error inflation from pressure check (due to updates from GMAO)

- Add YAMLs under config (End-to-End)
   This will be tested during the code sprint. 

- Add handling for CI tests with and without QC

To closely replicate the error assignment from the GSI error table. The ObsFunctionModelStepwiseLinear needs to add the capability of rounding the output of this function to the nearest integer. A related https://github.com/JCSDA-internal/ufo/pull/3085 has been submitted in the UFO repository.

Notes for test data set:
- The error inflation for duplicate observation is turned on for the UFO evaluation.
- The input obs file is a combined data type from 180 183 (SFCSHIP), 181, 187 (SFC), and 120 (Radiosonde Ps) for stationPressure.
- The GSI input obs error read from the prepbufr file is stored in the obs file for reference and validation. This can be used to validate the obs error values extracted from the errtable using ObsErrorModelStepwiseLinear function.

Please see [T2O Issue #96](https://github.com/NOAA-EMC/JEDI-T2O/issues/96) for documentation and UFO evaluation test results.

The CI test for conv_ps should change to green after merging this PR.

### CODE REVIEW

File: ush/ufoeval/run_ufo_hofx_test.sh
Comment (emilyhcliu) [inline ush/ufoeval/run_ufo_hofx_test.sh]: @CoryMartin-NOAA This is a temporary fix.  Please advise if you have any suggeston.  Thanks.

Comment (CoryMartin-NOAA) [reply ush/ufoeval/run_ufo_hofx_test.sh]: @emilyhcliu is this making the window larger because of the convention difference between JEDI and GSI? 
If so,
try:
```
window shift: True
```
after line 190

**Review by nicholasesposito — DISMISSED** [review]

File: ush/ufoeval/run_ufo_hofx_test.sh
Comment (emilyhcliu) [reply ush/ufoeval/run_ufo_hofx_test.sh]: @CoryMartin-NOAA Yes.  The hack is used due to the convention difference between JEDI and GSI.
Your suggestion worked.  Thanks a lot!!
I will modify it accordingly.

**Review by CoryMartin-NOAA — DISMISSED** [review]

**Comment by CoryMartin-NOAA** [comment]
@emilyhcliu can you resolve the conflicts and merge in develop, then I think this is ready to go! (well once the JCSDA merges the UFO pr)

**Comment by emilyhcliu** [comment]
> @emilyhcliu can you resolve the conflicts and merge in develop, then I think this is ready to go! (well once the JCSDA merges the UFO pr)

@CoryMartin-NOAA Done:  Updated the branch, resolved the conflict, and did git push.

**Review by CoryMartin-NOAA — DISMISSED** [review]
Approved, will not merge until UFO PR is merged

**Review by PraveenKumar-NOAA — DISMISSED** [review]
Looks good to me, except one general question: why surface_geometric_height is replaced by surface_altitude for geovar_sfc_geomz?

@PraveenKumar-NOAA Thanks for reviewing this PR.
You can use either one.  
The surface_altitude is the default and then GMAO added a second default (surface_geometric_height)

**Review by PraveenKumar-NOAA — DISMISSED** [review]
Looks good.

**Comment by emilyhcliu** [comment]
@CoryMartin-NOAA I updated the [test_yamls.sh ](https://github.com/NOAA-EMC/GDASApp/blob/8db6ab17737585a68eeddce151d460ea5f50d29e/ush/ufoeval/run_ufo_hofx_test.sh)to run CI test with and without QC.

**Review by CoryMartin-NOAA — APPROVED** [review]
Looks good, thanks @emilyhcliu!

**Review by ADCollard — APPROVED** [review]

**Comment by emilyhcliu** [comment]
@CoryMartin-NOAA The [UFO PR](https://github.com/JCSDA-internal/ufo/pull/3085) related to this PR has been approved and merged into UFO.  We are good to merge if Mr. Goofy (@ADCollard ) agrees with the changes in test_yaml.sh.

**Comment by CoryMartin-NOAA** [comment]
image

**Pull request merged**
